iquip/apps/scheduler.py

```diff
@@ -219,10 +217,62 @@ def paint(self,
     def sizeHint(self, option: QStyleOptionViewItem, index: QModelIndex) -> QSize:  # pylint: disable=unused-argument
         """Overridden."""
         data = index.data(Qt.DisplayRole)
+        if data is None:
+            return QSize(0, 0)
         experimentView = ExperimentView(data)
         return experimentView.sizeHint()
 
 
+class _ExperimentQueueFetcherThread(QThread):
+    """QThread for fetching the queued experiments from the proxy server.
+
+    Signals:
+        fetched(experimentList, runningExperiment):
+          The experiment queue and currently running experiment are fetched.
+    """
+
+    fetched = pyqtSignal(list, object)
+
+    def __init__(
+        self,
+        callback: Callable[[SubmittedExperimentInfo, None], List[SubmittedExperimentInfo]],
+        parent: Optional[QObject] = None
+    ):
+        """Extended.
+
+        Args:
+            callback: The callback method called after this thread is finished.
+        """
+        super().__init__(parent=parent)
+        self.fetched.connect(callback, type=Qt.QueuedConnection)
+
+    def run(self):
+        """Overridden.
+
+        Fetches the experiment list as a dictionary from the proxy server,
+        and emits a list and an ExperimentInfo instance for display.
+        """
+        while True:
+            try:
+                response = requests.get("http://127.0.0.1:8000/experiment/queue/", timeout=10)
+                response.raise_for_status()
+                response = response.json()
+            except requests.exceptions.RequestException as err:
+                print(err)
+                return
```

Comment:
Since you gave `timeout`, it could raise a timeout error.
Does the proxy server replies only when the queue status is changed?
If so, the timeout error may occur even if it is not an erroneous case.
Then it should `continue` instead of `return`.
Please confirm if I understand the proxy server's behavior correctly.

Reply:
Oh, you are right. I will change it that way.